src/scss/06-blocks/core/_heading.scss

```diff
@@ -1,7 +1,52 @@
-@for $i from 1 through 6 {
+/**
+ * Headings
+ *
+ * examples :
+ *  $headings: (
+ *      "1": (
+ *          "numbered"
+ *      ),
+ *      "2": (
+ *          "numbered",
+ *          "regular"
+ *      ),
+ *      "3": (
+ *          "numbered",
+ *          "regular"
+ *      ),
+ *      "4": (
+ *           "numbered",
+ *          "regular"
+ *      ),
+ *      "5": (
+ *          "numbered"
+ *      ),
+ *      "6": (
+ *          "numbered"
+ *      ),
+ *  );
+ *
+ */
+ 
+$headings: (
+    "1": (),
+    "2": (),
+    "3": (),
+    "4": (),
+    "5": (),
+    "6": (),
+);
+
+@each $i, $styles in $headings {
     h#{$i},
     .is-style-h#{$i} {
         @include heading(h#{$i});
+
+        @each $style in $styles {
```

Comment:
à ne mettre que sur les hn (car is-style-... ne sont pas cumulables dans gutenberg)